unit-tests-java8/src/test/java/org/eclipse/collections/test/bimap/BiMapTestCase.java

```diff
@@ -11,17 +11,101 @@
+    {
+        return false;
+    }
+
+    @Test
+    default void Iterable_sanity_check()
+    {
+        // Intentionally blank
+    }
+
+    @Override
+    default void Iterable_toString()
+    {
+        RichIterableUniqueTestCase.super.Iterable_toString();
+
+        BiMap<String, Integer> bimap = this.newWithKeysValues("Two", 2, "One", 1);
+        Assert.assertEquals("{Two=2, One=1}", bimap.toString());
+        Assert.assertEquals("[Two, One]", bimap.keysView().toString());
+        Assert.assertEquals("[2, 1]", bimap.valuesView().toString());
+        Assert.assertEquals("[Two:2, One:1]", bimap.keyValuesView().toString());
+        Assert.assertEquals("[2, 1]", bimap.asLazy().toString());
+
+        Assert.assertEquals(
+                "{10=4, 9=4, 8=4, 7=4, 6=3, 5=3, 4=3, 3=2, 2=2, 1=1}",
+                this.newWith(4, 4, 4, 4, 3, 3, 3, 2, 2, 1).toString());
+        Assert.assertEquals(
+                "[10, 9, 8, 7, 6, 5, 4, 3, 2, 1]",
+                this.newWith(4, 4, 4, 4, 3, 3, 3, 2, 2, 1).keysView().toString());
+        Assert.assertEquals(
+                "[4, 4, 4, 4, 3, 3, 3, 2, 2, 1]",
+                this.newWith(4, 4, 4, 4, 3, 3, 3, 2, 2, 1).valuesView().toString());
+        Assert.assertEquals(
+                "[10:4, 9:4, 8:4, 7:4, 6:3, 5:3, 4:3, 3:2, 2:2, 1:1]",
+                this.newWith(4, 4, 4, 4, 3, 3, 3, 2, 2, 1).keyValuesView().toString());
+    }
+
+    @Override
+    default void MapIterable_forEachKeyValue()
+    {
+        BiMap<Object, Integer> bimap = this.newWith(3, 2, 1);
+        MutableCollection<Integer> forEachKeyValue = this.newMutableForFilter();
+        bimap.forEachKeyValue((key, value) -> forEachKeyValue.add(value + 10));
+        assertEquals(this.newMutableForFilter(13, 13, 13, 12, 12, 11), forEachKeyValue);
+
+        BiMap<Integer, String> bimap2 = this.newWithKeysValues(3, "Three", 2, "Two", 1, "One");
+        MutableCollection<String> forEachKeyValue2 = this.newMutableForFilter();
+        bimap2.forEachKeyValue((key, value) -> forEachKeyValue2.add(key + value));
+        assertEquals(this.newMutableForFilter("3Three", "2Two", "1One"), forEachKeyValue2);
+
+        MutableCollection<Integer> forEachValue = this.newMutableForFilter();
+        bimap.forEachValue(value -> forEachValue.add(value + 10));
+        assertEquals(this.newMutableForFilter(13, 13, 13, 12, 12, 11), forEachValue);
+
+        MutableCollection<Object> forEachKey = this.newMutableForFilter();
+        bimap2.forEachKey(key -> forEachKey.add(key + 1));
+        assertEquals(this.newMutableForFilter(4, 3, 2), forEachKey);
+    }
+
+    @Override
+    default void MapIterable_flipUniqueValues()
+    {
+        BiMap<String, Integer> bimap = this.newWithKeysValues("Three", 3, "Two", 2, "One", 1);
+        BiMap<Integer, String> result = bimap.flipUniqueValues();
+
+        assertEquals(
+                this.newWithKeysValues(3, "Three", 2, "Two", 1, "One"),
+                result);
+    }
```

Comment:
This override is because the superclass tests that an assertion is thrown when flipping a map with duplicate values. A BiMap cannot have duplicate values and never throws from flipUniqueValues().